src/timer.rs

```diff
@@ -279,9 +279,9 @@ where
         let timeout: Self::Time = timeout.into();
         let clock = TIM::clock(&self.clocks);
 
-        let ticks = clock.integer().saturating_mul(timeout.integer()) * *timeout.scaling_factor();
+        let ticks = (clock.integer() as u64).saturating_mul(timeout.integer() as u64) * *timeout.scaling_factor();
```

Comment:
This is definitely a good improvement, please use `u64::from` instead of as casts :) 

```suggestion
        let ticks = u64::from(clock.integer()).saturating_mul(u64::from(timeout.integer()) * *timeout.scaling_factor();
```